src/rgw/rgw_lc.cc

```diff
@@ -1175,7 +1175,7 @@ class LCOpAction_CurrentExpiration : public LCOpAction {
 		       << " " << oc.wq->thr_name() << dendl;
     } else {
       /* ! o.is_delete_marker() */
-      r = remove_expired_obj(oc.dpp, oc, !oc.bucket->versioned(),
+      r = remove_expired_obj(oc.dpp, oc, oc.bucket->versioned(),
```

Comment:
This is not correct. This will then end up not removing objects when the bucket is not versioned. I think the right fix is to check if versioning is in enabled state.. i.e,

remove_expired_obj(oc.dpp, oc, !oc.bucket->versioning_enabled(),...)

Reply:
Why is there this MR: BUG about lifecycle: https://tracker.ceph.com/issues/65772. 
After I retested, there was only a bug regarding CurrentExpiration with Bucket-suspend-versioning (version_id=null), while other lifecycle cases still functioned normally.

Reply:
If there is no delete marker -> Go to the else branch to proceed with checking and deleting the object. There will be two cases here: bucket-non-versioning & bucket_version (enable versioning || suspend versioning)

- For Bucket-non-versioning: !oc.bucket->versioned() returns 1 (remove_indeed=True).

- For Bucket-versioned: !oc.bucket->versioned() returns 0 (remove_indeed=False) -> The function remove_expired_obj does not delete the object immediately but goes through processing steps (Checked the log and saw the log DELETED, but the object was not deleted -> Only notify was executed).

=> Solution: Reverse the condition.

- For Bucket-non-versioning: Perform current expiration without checking any conditions and delete the object immediately -> therefore, set remove_indeed = False.

- For Bucket-versioned: Perform current expiration -> set the condition remove_indeed = True -> Proceed to delete objects that satisfy the lifecycle.

Reply:
As per AWS spec, https://docs.aws.amazon.com/AmazonS3/latest/userguide/lifecycle-expire-general-considerations.html , if the objects are expired in a version enabled bucket, those objects are not deleted but are made non-current after creating a delete_marker. So "remove_indeed" should be set to False in such cases.

So the only bug in the current code is when the bucket is versioned but is suspended in which case objects should be immediately deleted (like in case of non-versioned buckets).

Reply:
Yes, I am encountering a bug with the Bucket Suspend Versioning case, which is why this PR exists.

Reply:
but with the current fix posted, `remove_indeed` will be set to  `false` in the case of non-versioned buckets. Some SALs may rely on that flag and it could result in not deleting objects in those buckets .

Hence i am suggesting to modify that check as below --

`r = remove_expired_obj(oc.dpp, oc, !oc.bucket->versioning_enabled(),...`

This will set `remove_indeed = false` only when versioning is in `Enabled` state.

Reply:
Thank you for your suggestion. I have rerun the test, and the result was successful.

Reply:
@mattbenjamin Can you review my PR? Thanks

Reply:
@cbodley @mattbenjamin Can you review help me ? Thanks